xapian-bindings/python3/pythonletortest.py

```diff
@@ -0,0 +1,184 @@
+    import filecmp
+    expect(filecmp.cmp(training_data, "training_output1.txt"), True)
+    os.remove("training_output1.txt")
+
+def test_ranker():
+    ranker = xapianletor.ListNETRanker()
+    expect_exception(xapianletor.FileNotFoundError, "No training file found. Check path.", ranker.train_model, "")
+    setup_database_two()
+    dbpath = b"database_two_docs"
+    enquire = xapian.Enquire(xapian.Database(dbpath))
+    enquire.set_query(xapian.Query("lions"))
+    mset = enquire.get_mset(0, 10)
+    expect(mset.size(), 2)
+
+    data_directory = "./testdata/"
+    query = data_directory + "query.txt"
+    qrel = data_directory + "qrel.txt"
+    training_data = data_directory + "training_data.txt"
+    ranker.set_database_path(dbpath)
+    expect(ranker.get_database_path(), dbpath)
+
+    ranker.set_query(xapian.Query("lions"))
+    ranker.train_model(training_data, "ListNet_Ranker")
+    doc1 = mset[0]
+    doc2 = mset[1]
+    ranker.rank(mset, "ListNet_Ranker")
+    expect(doc1.docid, mset[1].docid)
+    expect(doc2.docid, mset[0].docid)
+    expect_exception(xapianletor.LetorInternalError, None,
+                     ranker.score, query, qrel, "ListNet_Ranker",
+                     "scorer_output.txt", 10, "")
+    expect_exception(xapianletor.FileNotFoundError, None,
+                     ranker.score, "", qrel, "ListNet_Ranker",
+                     "scorer_output.txt", 10)
+    expect_exception(xapianletor.FileNotFoundError, None,
+                     ranker.score, qrel, "", "ListNet_Ranker",
+                     "scorer_output.txt", 10)
+    ranker.score(query, qrel, "ListNet_Ranker",
+                 "ndcg_output_ListNet_2.txt", 10)
+    expect(os.path.isfile("ndcg_output_ListNet_2.txt"), True)
+    os.remove("ndcg_output_ListNet_2.txt")
+    ranker.score(query, qrel, "ListNet_Ranker",
+                 "err_output_ListNet_2.txt", 10, "ERRScore")
+    expect(os.path.isfile("err_output_ListNet_2.txt"), True)
+    os.remove("err_output_ListNet_2.txt")
+
+def test_import_letor_star():
+    """Test that "from xapianletor import *" works.
+
+    It's not normally good style to use it, but it should work anyway!
+
+    """
+    import test_xapian_letor_star
+
+result = True
+
+# Run all tests (ie, callables with names starting "test_").
+def run():
+    global result
+    if not runtests(globals(), sys.argv[1:]):
+        result = False
+
+print("Running tests without threads")
+run()
+
+if have_threads:
+    print("Running tests with threads")
+
+    # This testcase seems to just block when run in a thread under Python 3
+    # on some plaforms.  It fails with 3.2.3 on Debian wheezy, but passes
+    # with the exact same package version on Debian unstable not long after
+    # the jessie release.  The issue it's actually serving to regression
+    # test for is covered by running it without threads, so just disable it
+    # rather than risk test failures that don't seem to indicate a problem
+    # in Xapian.
```

Comment:
Just copying this comment over results in it making some rather misleading claims - I doubt test_import_letor_star has really been tested on Debian wheezy or jessie at all.

Reply:
fixed in a fixup commit.